reactor-core/src/main/java/reactor/core/publisher/Flux.java

```diff
@@ -3184,23 +3184,25 @@ public final T blockLast(Duration timeout) {
 				catch (Exception e) {
 					return Mono.error(e);
 				}
-				if (v == null) {
-					return Mono.onAssembly(new MonoSupplier<>(listSupplier()));
-				}
-				return Mono.just(v).map(u -> {
+				return Mono.fromCallable(() -> {
```

Comment:
I would suggest to use the form `onAssembly(new MonoCallable<>(...))` here and below, so that there is no additional step in the backtrace that corresponds to the `fromCallable` call.